src/plugins/intel_npu/src/plugin/npuw/base_sync_infer_request.cpp

```diff
@@ -243,14 +243,27 @@ void ov::npuw::IBaseInferRequest::infer() {
     m_now_idx.reset();
     prepare_for_infer();
     bool failover_happened = false;
+    double sdpa_time = 0.0;
+    double gemm_time = 0.0;
+    auto t_start = std::chrono::high_resolution_clock::now();
     for (std::size_t idx = 0u; idx < m_num_submodels; idx++) {
         m_now_idx = idx;
         if (!valid_subrequest(idx)) {
             continue;
         }
         subscribe_subrequest(idx, [](std::exception_ptr) {});
         bool failover = false;
+        auto t_start = std::chrono::high_resolution_clock::now();
         run_subrequest_for_success(idx, failover);
+        auto t_end = std::chrono::high_resolution_clock::now();
+
+        double elapsed_time_ms = std::chrono::duration<double, std::milli>(t_end - t_start).count();
+        auto& comp_model_desc = m_npuw_model->m_compiled_submodels[idx];
+        if (comp_model_desc.attention) {
+            sdpa_time += elapsed_time_ms;
+        } else {
+            gemm_time += elapsed_time_ms;
+        }
```

Comment:
we have a proper profiling utilities here https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_npu/src/plugin/npuw/perf.hpp

so this code could look like

```suggestion
        auto ms = ms_to_run([&](){ run_subrequest_for_success(idx, failover); });
        
        auto& comp_model_desc = m_npuw_model->m_compiled_submodels[idx];
        (comp_model_desc.attention ? sdpa_time : gemm_time) += elapsed_time_ms;
```

Where sdpa_time and gemm_time could be declared as `perf::metric<float, MSec>` so you get the statistics collected for free